reanalysis9

```diff
@@ -6,206 +6,216 @@
-total.new.temp = total.new[-which(total.new$icd10_date == 'NA' | total.new$mi_date == 'NA'), ]
-total.new.temp$icd10[as.Date(total.new.temp$icd10_date) > as.Date("2015-03-31")] = "NA"
-total.new.temp$icd10[as.Date(total.new.temp$icd10_date) < as.Date("1998-01-01")] = "NA"
-total.new.temp$icd10[as.Date(total.new.temp$mi_date) - as.Date(total.new.temp$icd10_date) < 8] = "NA"
-total.new = rbind(total.new.temp, total.new.na)
-rm(total.new.temp, total.new.na)
+to_remove <- total.new$eid[total.new$mi_date > as.Date("2015-03-01", "%Y-%m-%d") | total.new$mi_date < total.new$startfollowup]
+to_remove <- unique(to_remove[!is.na(to_remove)])
+
+total.new = total.new[!total.new$eid %in% to_remove,]
 nrow(total.new)
-#2728566 rows
+#2816591 rows
 length(unique(total.new$eid))
-#392743 individuals
+#495453 individuals
 length(unique(total.new$icd10))
-#1384 ICD codes
-length(unique(total[which(total$mi_date != "NA"), ]$eid))
-#17620 individuals with MI
-
-# remove individuals with I25
-total.new = total.new[!total.new$icd10 == "I25", ]
+#1393 ICD codes
+length(unique(total.new[which(!is.na(total.new$mi_date)), ]$eid))
+#14584 individuals with MI
+
+# set to missing ICD10 codes/dates outside of timeframe or that are before 7 days from MI date
+total.new$icd10[total.new$icd10_date > as.Date("2015-03-01", "%Y-%m-%d") | total.new$icd10_date < total.new$startfollowup] <- NA
+total.new$icd10_date[total.new$icd10_date > as.Date("2015-03-01", "%Y-%m-%d") | total.new$icd10_date < total.new$startfollowup] <- NA
+total.new$icd10[(total.new$mi_date - total.new$icd10_date) < 8] <- NA
+total.new$icd10_date[(total.new$mi_date - total.new$icd10_date) < 8] <- NA
 nrow(total.new)
-#2702582 rows
+#2816591 rows
 length(unique(total.new$eid))
-#391918 individuals
+#495453 individuals
 length(unique(total.new$icd10))
-#1383 ICD codes
-length(unique(total[which(total$mi_date != "NA"), ]$eid))
-#17620 individuals with MI
-
-# make binary variable for myocardial infarction (e.g., 0 is no MI and 1 is MI)
-total.new.mi = total.new[which(total.new$mi_date != 'NA'), ]
-total.new.no.mi = total.new[which(total.new$mi_date == 'NA'), ]
-total.new.mi = cbind(total.new.mi, "mi" = "1")
-total.new.no.mi = cbind(total.new.no.mi, "mi" = "0")
-total.new = rbind(total.new.mi, total.new.no.mi)
-rm(total.new.mi, total.new.no.mi)
-
-# exclude ICD10 codes after first diagnosis
-total.new = total.new[order(total.new$icd10), ]
-total.new = total.new[order(total.new$eid), ]
-total.new = total.new[!duplicated(total.new[, c("eid", "icd10")], fromLast = FALSE), ]
+#1385 ICD codes
+length(unique(total.new[which(!is.na(total.new$mi_date)), ]$eid))
+#14584 individuals with MI
+
+# remove individuals with I25 before MI
+# LET'S CHECK WITH ERIK IF THIS EXCLUSION IS NEED IT
+to_remove <- total.new$eid[total.new$icd10 == "I25"]
+to_remove <- unique(to_remove[!is.na(to_remove)])
+total.new = total.new[!total.new$eid %in% to_remove, ]
 nrow(total.new)
```

Comment:
Remove Individuals that have an I25 before MI. We need to ask Erik if this makes sense